src/fast-copy.cpp

```diff
@@ -1,123 +1,108 @@
 // These routines were adapted from Kevin Ushey's code in hadley/reshape
 
 #include <Rcpp.h>
+#include "utils.h"
 
 using namespace Rcpp;
 
 
-#define DO_REP_EACH_N(RTYPE, CTYPE, ACCESSOR)          \
-  {                                                    \
-    int counter = 0;                                   \
-    Shield<SEXP> out(Rf_allocVector(RTYPE, out_size)); \
-    CTYPE* x_ptr = ACCESSOR(x);                        \
-    CTYPE* out_ptr = ACCESSOR(out);                    \
-    for (int i = 0; i < x_size; ++i) {                 \
-      for (int j = 0; j < times[i]; ++j) {             \
-        out_ptr[counter] = x_ptr[i];                   \
-        ++counter;                                     \
-      }                                                \
-    }                                                  \
-    return out;                                        \
-    break;                                             \
+#define DO_REP_EACH_N(RTYPE, CTYPE, ACCESSOR)           \
```

Comment:
Are the changes here just whitespace? (should try to avoid whitespace-only changes since they can make pull requests harder to review)